tests/cache_test.py

```diff
@@ -172,3 +171,40 @@ def lambda_handler(event, context):
             return 'Hello from Lambda with secret %s' % secret_value
         
         lambda_handler(None, None)
+
+    # This isn't any different than the default behaviour, but allows
+    # us to confirm that the code calls the provided factory method.
+    def _get_client(self, session, name):
+        self.got_client = True
+        return session.client(name)
+
+    # This isn't any different than the default behaviour, but allows
+    # us to confirm that the code calls the provided factory method.
+    def _get_session(self):
+        self.got_session = True
+        return boto3.session.Session()
+
+    def test_session_factory(self):
+        # Clear the cached session and client from previous tests
+        SSMParameter.BOTO3_SESSION = None
+        SSMParameter.BOTO3_CLIENTS = {}
+
+        # Set the session and client factories
+        SSMParameter.BOTO3_SESSION_FACTORY = self._get_session
```

Comment:
Would you mind adding a clear example of this in the README file as well? :)

Reply:
Can do. Let me know what you think about the idea for resolving #3 that I wrote at the top -- I think that obsoletes this solution and would change the example, so wouldn't want to duplicate work. I'm hoping to have the strawman ready for comment later today.

Reply:
Done, I hope.